### DISCUSSION TITLE
Yaskawa Descriptions in ROS 2

### DISCUSSION THREAD

**Discussion by destogl** [Yaskawa-Global/motoros2#161] [General]
Hi all,

we are currently integrating Yaskawa GP250 in a project and things went very smooth. Thanks for the great manual!

During the process, some questions arose that I would like to clarify with you:

1. From where should we use robot descriptions for the robots? We are currently using [motoman](https://github.com/gavanderhoorn/motoman/tree/ros2) repository from @gavanderhoorn which we extended with GP250 and GP280 kinematics (first we got information that robot is 280).
2. We didn't add meshes into those repositories because of the restrictive license. How did you handle this for other robots?
3. We had some issues with the default joint names. Are those changed between `motoros` and `motoros2`?  am asking because many robots have suffixes, e.g. "_s", "_r" and so on, in their descriptions. Would it be possible to unify those naming (in ROS 2 version of descriptions) to the default names using in ROS 2 driver, i.e., "joint_1", "joint_2", etc?
4. We have also created a generic MoveIt 2 configuration that can be used on all robots in the [motoman](https://github.com/gavanderhoorn/motoman/tree/ros2) repository, considering that joint names are unified. Where should we submit this?
5. Furthermore, something that might not be so relevant to the driver, but it is very useful for development, we have created a ros2_control support package that enables mocking of hardware. This approach we use very frequently in the project to set up the whole pipeline with MoveIt and environment before moving to the real hardware. The configuration is set in such a way that w can easily switch between mocked HW and real hardware. Moreover, this opens the doors toward simulators supported with ros2_control without big effort. This might be even a separate topic, we are glad to discuss and contribute, if you think that it is useful for the broader community.

**Comment by ted-miller** [comment]
Hi @destogl 

1. Yes, that is currently the best repository for this.
2. Can you clarify about the restrictive license? These seem to all be BSD which AFAIK is not restrictive. (Admittedly, I struggle to understand the nuance of some licenses.)
3. I agree that these should be updated before we merge gavanderhoorn's fork into this repository.
4. That's great. I think a subfolder in the root of this repository would be a good spot. If you could submit a pull request it would be appreciated. (I personally can't mentally keep track of any more repos.)
5. I'll let @gavanderhoorn comment on this topic.

**Reply by ted-miller** [reply]
4. Actually... it probably doesn't belong in there. We really need to get [this](https://github.com/gavanderhoorn/motoman/tree/ros2) polished up and ported to Yaskawa-global. Then we should probably include your stuff in there.

**Reply by destogl** [reply]
yes, this is where we already placed it. After current testing on the hardware, we will open PR(s) for it on the @gavanderhoorn's motoman repository.

**Comment by destogl** [comment]
Hi @ted-miller thanks for the fast answer,

what I mean is that the license on Yaskawa page for downloading meshes is not open-source friendly. The code in the `motoman` repository is fine. [This is the page](https://www.motoman.com/en-us/products/robots/industrial/assembly-handling/gp-series/gp250) where we were first looking at. I just saw that [on the German page one](https://www.yaskawa.de/produkte/roboter/handling-montage/productdetail/product/gp250_697) can download CAD files without any problems, but still would like to check the license with you and if may generate meshes from that CAD files and add them to the repository.

Regarding other point, we will work on cleaning and opening PR for all the points.

**Reply by ted-miller** [reply]
@EricMarcil, can you comment on this?

**Reply by EricMarcil** [reply]
I usually make most of the URDF and post them.  I use the MotoSim model and convert them to .dea or .stl
The MotoSim model are usually lighter than the fully detailled 3D models from those sites.
I saw the "Terms of Use and Copyright Notice" on the page that he is refering to.  I don't think it is intended to block "good" use of the downloaded material but I'm not sure of the legal aspects of it.  We would need to verify internally.

**Reply by EricMarcil** [reply]
@destogl If you are looking for a specific model.  Let me know, which one.

**Reply by destogl** [reply]
@EricMarcil we are looking for GP250 and GP280

**Reply by destogl** [reply]
Are there MotoSim models for download?

**Reply by EricMarcil** [reply]
@destogl No, they are included in the MotoSim software (that is a purchase software).  The file format is also not very common HOOPS .hsf.  I have an application that converts them to .dae.  I've converted the base models but noticed that the side cables and brackets models is not include.  They are modelled differently and would need to be manually added.

I can send you the base models if you are in a rush.  Or if you can wait a bit, I should be able to create the URDF for those models early next week.

**Reply by destogl** [reply]
Hi @EricMarcil we have started to create full models from the CAD files. We already have URDFs, but we didn't have meshes because of license discussion. We have already pipeline to do that and If this is OK with you, we can finish this up. Expect you would like to have the models from the simulator.

**Reply by EricMarcil** [reply]
@destogl If you are worried about the license and you don't mind.  Send me the URDF you have and model.  I'll review it, if needed swap out the mesh, and publish it.  I'll giving you the credit for it and I'll be responsible for publishing it, so won't have to worry about any license issue.

**Reply by EricMarcil** [reply]
Send it to:
eric.marcil@motoman.com
